Feat: Custom plugins reference (#1250)
Kong/developer.konghq.com: app/custom-plugins/reference.md

```
kong.plugins.my-custom-plugin.<module_name>
```
{:.no-copy-code}

### Basic plugin modules

In its simplest form, a plugin consists of two mandatory modules:
* `handler.lua`: This module is the core of your plugin. It's an interface to implement, in which each function will be run at the desired moment in the lifecycle of a request/connection.
* `schema.lua`: This module holds the schema of the plugin configuration to be entered by the user, and defines rules on it, so that the user can only enter valid configuration values.

Some plugins, such as the [File Log](https://github.com/Kong/kong/tree/master/kong/plugins/file-log) plugin, use only these two modules. However, you can use advanced modules to implement extra functionalities.

To learn how to create a simple plugin with the basic modules, see the [getting started guide](/custom-plugins/#get-started).

### Advanced plugin modules

Some plugins might have to integrate deeper with {{site.base_gateway}}: have their own table in the database, expose endpoints in the Admin API, etc. 
This can be done by adding new modules to your plugin. 

> Here is what the structure of a plugin looks like when it implements all of the optional modules:

Reviewer: ```suggestion
Below is the structure of a plugin with all optional modules implemented:
```

```
complete-plugin
├── api.lua
├── daos.lua
├── handler.lua
├── migrations
│   ├── init.lua
│   └── 000_base_complete_plugin.lua
└── schema.lua
```
{:.no-copy-code}

> Here is the complete list of possible modules to implement:

Reviewer: ```suggestion
The complete list of modules you can implement is as follows:
```
